### PR TITLE
[11.0] Add module stock picking restrict cancel with original move

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

File: stock_picking_restrict_cancel_with_orig_move/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Picking Restrict Cancel with Original Moves",
+    "summary": "Restrict cancellation of dest moves according to origin.",
+    "version": "11.0.1.0.0",
+    "category": "Warehouse",
+    "website": "https://github.com/OCA/stock-logistics-workflow",
+    "author": "Camptocamp SA, Odoo Community Association (OCA)",
```
Comment (simahawk) [inline stock_picking_restrict_cancel_with_orig_move/__manifest__.py]: remove SA

File: stock_picking_restrict_cancel_with_orig_move/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class StockMove(models.Model):
+
+    _inherit = 'stock.move'
+
+    def _action_cancel(self):
+        if (all(move.state in ('cancel', 'done')
+                for move in self.mapped('move_orig_ids'))
+                or self.env.context.get('bypass_check_state')):
```
Comment (simahawk) [inline stock_picking_restrict_cancel_with_orig_move/models/stock_move.py]: ctx key check should the 1st thing to do to avoid all the rest if not needed.

**Review by simahawk — CHANGES_REQUESTED** [review]
LGTM, small things

**Review by tschanzt — APPROVED** [review]

**Review by jgrandguillaume — APPROVED** [review]
LGMT ! Thanks (fct + code review)

**Comment by pedrobaeza** [comment]
Merging this module, but this maybe can be proposed at least for master for fixing the behavior upstream.

**Pull request merged**